guessing_game.rb

```diff
@@ -0,0 +1,81 @@
+require 'pry-byebug'
+
+class GuessingGame
+
+  attr_accessor :congrats_message, :remaining_guesses, :won
+
+  def initialize(secret_number, num_guesses)
+    @congrats_message = "Yay, you won!"
+    @secret_number = secret_number
+    @remaining_guesses = num_guesses
+    @guessed_numbers = []
```

Comment:
it will be faster to store the previous guesses in a hash. you can access a hash element in `O(1)` constant time, whereas searching an array for a particular element will require `O(n)` linear search